Index: ssl.py
===================================================================
--- ssl.py	(revision 1791912)
+++ ssl.py	(working copy)
@@ -342,6 +342,9 @@
 Purpose.CLIENT_AUTH = Purpose('1.3.6.1.5.5.7.3.2')
 
 
+_certificates = {}
+
+
 class SSLContext(_SSLContext):
     """An SSLContext holds various SSL-related configuration options and
     data, such as certificates and possibly a private key."""
@@ -392,14 +395,17 @@
 
     def _load_windows_store_certs(self, storename, purpose):
         certs = bytearray()
-        try:
-            for cert, encoding, trust in enum_certificates(storename):
-                # CA certs are never PKCS#7 encoded
-                if encoding == "x509_asn":
-                    if trust is True or purpose.oid in trust:
-                        certs.extend(cert)
-        except OSError:
-            warnings.warn("unable to enumerate Windows certificate store")
+        if _certificates.get(storename) is None:
+            try:
+                _certificates[storename] = enum_certificates(storename)
+            except OSError:
+                warnings.warn("unable to enumerate Windows certificate store")
+                _certificates[storename] = []
+        for cert, encoding, trust in _certificates[storename]:
+            # CA certs are never PKCS#7 encoded
+            if encoding == "x509_asn":
+                if trust is True or purpose.oid in trust:
+                    certs.extend(cert)
         if certs:
             self.load_verify_locations(cadata=certs)
         return certs
